lib/ansible/modules/network/junos/junos_config.py

```diff
@@ -295,7 +295,7 @@ def main():
         # deprecated replace in Ansible 2.3
         replace=dict(type='bool'),
 
-        confirm=dict(default=0, type='int'),
+        confirm=dict(default=0, type='str'),
```

Comment:
As the input value is supposed to be an integer changing `int` to `str` doesn't seem right. Since the underlying ncclient API is expecting value of timeout in string format how about changing to string it while passing the arguments?
```
$ git diff lib/ansible/modules/network/junos/junos_config.py
diff --git a/lib/ansible/modules/network/junos/junos_config.py b/lib/ansible/modules/network/junos/junos_config.py
index 894d849..a444672 100644
--- a/lib/ansible/modules/network/junos/junos_config.py
+++ b/lib/ansible/modules/network/junos/junos_config.py
@@ -184,7 +184,7 @@ from ansible.module_utils.network.junos.junos import get_diff, load_config, get_
 from ansible.module_utils.network.junos.junos import commit_configuration, discard_changes, locked_config
 from ansible.module_utils.network.junos.junos import junos_argument_spec, load_configuration, get_connection, tostring
 from ansible.module_utils.six import string_types
-from ansible.module_utils._text import to_native
+from ansible.module_utils._text import to_native, to_text

 try:
     from lxml.etree import Element, fromstring
@@ -365,7 +365,7 @@ def main():
                         if module.params['confirm'] > 0:
                             kwargs.update({
                                 'confirm': True,
-                                'confirm_timeout': module.params['confirm']
+                                'confirm_timeout': to_text(module.params['confirm'], errors='surrogate_then_replace')
                             })
                         commit_configuration(module, **kwargs)
                     else:
```